src/gt4py/backend/dawn_backends.py

```diff
@@ -401,8 +401,12 @@ def backend_imports(self) -> List[str]:
         return ["import cupy"]
 
     def backend_pre_run(self) -> List[str]:
-        field_names = self.args_data["field_info"].keys()
-        return ["{name}.host_to_device()" for name in field_names]
+        field_names = [
+            key
+            for key in self.args_data["field_info"]
+            if self.args_data["field_info"][key] is not None
+        ]
```

Comment:
I think this is slighty more readable:
```suggestion
        field_names = [
            key
            for key, value in self.args_data["field_info"].items()
            if value is not None
        ]
```